### PR TITLE
WebUI: Certificate login

### PR DESCRIPTION
**Pull request by pvomacka**
https://pagure.io/freeipa/issue/6225

### CODE REVIEW

**Comment by dkupka** [comment]
@pvomacka NACK, see lint errors in travis.

File: install/ui/src/freeipa/widgets/LoginScreen.js [hunk 1]
```diff
@@ -72,9 +78,12 @@ define(['dojo/_base/declare',
 
         user_locked: "The user account you entered is locked. ",
 
+        login_url: '/ipa/session/login_x509',
```
Comment (pvoborni) [inline install/ui/src/freeipa/widgets/LoginScreen.js]: this should be called e.g. `x509_url` login is too general.

**Comment by pvomacka** [comment]
@pvoborni  thank you for review. Fixed all proposed changes.

File: install/conf/ipa.conf [hunk 2]
```diff
@@ -97,6 +102,24 @@ Alias /ipa/session/cookie "/usr/share/ipa/gssapi.login"
...
+<Location "/ipa/session/login_x509">
+  AuthType none
+  Require all granted
+  GssapiCredStore keytab:/var/lib/ipa/gssproxy/http.keytab
+  GssapiCredStore client_keytab:/var/lib/ipa/gssproxy/http.keytab
+  GssapiDelegCcacheDir /var/run/ipa/ccaches
+  GssapiDelegCcachePerms mode:0660 gid:ipaapi
+  GssapiImpersonate On
+  NSSVerifyClient require
+  NSSOCSP On
```
Comment (dkupka) [inline install/conf/ipa.conf]: NSSOCSP can't be specified inside <location/>. That's why Travis crashes.

**Comment by dkupka** [comment]
LGTM and works.

**Comment by dkupka** [comment]
master:

* 75c592d3b9081474cae51c929e6af29c7a0eebb6 Support certificate login after installation and upgrade
* 585547ee9478ea0173106d88d40d7807baab8bcf WebUI: add link to login page which for login using certificate

**Comment by simo5** [comment]
NACK NACK NACK
Pleas revert the change to the gssproxy template, it undoes half the work done in privilege separation

**Comment by simo5** [comment]
You need to wait to get th gssproxy fix I've been developing today and set the minimum gssproxy version to the one with the fix once we get to publish it

**Comment by pvomacka** [comment]
Removed in https://github.com/freeipa/freeipa/pull/585 once it will be pushed I will close this one again.

**Comment by MartinBasti** [comment]
#585 was pushed

**Pull request closed without merging**